src/util/util.h

```diff
@@ -849,4 +850,10 @@ uint64_t get_spend_time_us(uint64_t st);
 #define CHECK_PAC_CHECK_UPN_ALLOW_MISSING (1 << 5)
 
 errno_t get_pac_check_config(struct confdb_ctx *cdb, uint32_t *pac_check_opts);
+
+static inline struct timeval sss_tevent_timeval_current_ofs_time_t(time_t secs)
+{
+    uint32_t secs32 = (secs > UINT_MAX ? UINT_MAX : secs & 0x0FFFFFFFF);
```

Comment:
IIUC, what you are saying is that, in the case where `secs` < `UINT_MAX`, we are only dropping 0 bits, so the `&` operation is useless. And you are right.

The reason we do this is that coverity was complaining for the automatic conversion. However, since we now have the conditional part, this could solve the problem and make the `&` useless.

Let me check and come back to you.